server/api.go

```diff
@@ -495,6 +505,116 @@ func (p *Plugin) getYourPrs(w http.ResponseWriter, r *http.Request) {
 	w.Write(resp)
 }
 
+func (p *Plugin) getPrsDetails(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := context.Background()
+
+	var githubClient *github.Client
+
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
+		return
+	} else {
+		githubClient = p.githubConnect(*info.Token)
+	}
+
+	var prList []*PRDetails
+	json.NewDecoder(r.Body).Decode(&prList)
+
+	var prDetails []*PRDetails
+
+	ch := make(chan *PRDetails)
+	for _, pr := range prList {
+		go fetchPRDetails(ctx, githubClient, pr.URL, pr.Number, ch)
+	}
+
+	for range prList {
+		pr := <-ch
+		prDetails = append(prDetails, pr)
+	}
+
+	resp, _ := json.Marshal(prDetails)
+	w.Write(resp)
+}
+
+func fetchPRDetails(ctx context.Context, client *github.Client, prURL string, prNumber int, ch chan *PRDetails) {
+	status := ""
+	requestedReviewers := []*string{}
+	var reviewsList []*github.PullRequestReview = nil
+
+	repoOwner, repoName := getRepoOwnerAndNameFromURL(prURL)
+
+	chReviews := make(chan []*github.PullRequestReview)
+
+	go fetchReviews(ctx, client, repoOwner, repoName, prNumber, chReviews)
```

Comment:
Similar to the above, `fetchReviews` should just return the reviews list: there's no need for a channel at all in this case. Use the goroutine for parallelism, but just write directly a local variable and coordinate with a `sync.WaitGroup`.